service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapperTest.java

```diff
@@ -80,14 +91,32 @@ public void setUp() {
             BUNDLE_ENTRY_WITH_CONSULTATION,
             BUNDLE_ENTRY_WITH_LOCATION
         ));
-        encounterMapper = new EncounterMapper(messageContext, encounterComponentsMapper);
+        encounterMapper = new EncounterMapper(messageContext, encounterComponentsMapper, confidentialityService);
     }
 
     @AfterEach
     public void tearDown() {
         messageContext.resetMessageContext();
     }
 
+    @Test
+    public void testEncounterWithNOPATAddsConfidentialityCodeIntoEhrComposition() {
+        lenient().when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
```

Comment:
not a change, just some info.  You only need the lenient() when there will be cases where mocked service will not occur, so for this test it shouldn't need to be included